src/SnowdriftEventHandler.hs

```diff
@@ -229,9 +229,36 @@ notificationEventHandler AppConfig{..} (EDeletedPledge _ user_id project_id _) =
             (\route -> userDisplayName (Entity user_id u)
                    <> " is no longer supporting the [project](" <> route <> ")")
 
-handleWatched :: Maybe NotificationSender -> Text -> ProjectId
-              -> (Text -> Route App) -> ProjectNotificationType
-              -> (Text -> Text) -> SDB ()
+notificationEventHandler AppConfig{..} (EVolunteerApp _ user_id project_id app_id) =
+    -- This pattern match will fail if the project is deleted from the database
+    -- by the time this event is handled. Not sure if that's a problem or not.
+    runSDB $ do
+        [project_name] <-
```

Comment:
Impartiality would still be the problem. This should be using `get` and Maybe (perhaps MaybeT), though that doesn't have to happen *today*.